File: backend/test-with-js/api-test/data/loadCSV.js
import fs from 'fs';
import csv from 'csv-parser';
import {createObjectCsvWriter} from 'csv-writer';

export async function readCSV(dataFileName, columnName = '', filterValue = '') {
    const env = process.env["env"] ? process.env["env"] : "dev";
    const filePath = './data/' + env + '/' + dataFileName
    try {
        return new Promise((resolve, reject) => {
            const results = [];
            const stream = fs.createReadStream(filePath).pipe(csv());
            stream.on('data', (row) => {
                if (!columnName || row[columnName] === filterValue) {
                    results.push(row);
                }
            });

            stream.on('end', () => resolve(results));
            stream.on('error', (error) => reject(error));
        });
    } catch (error) {
        console.log('Error:', error);
    }
}

export async function writeCSV(datafileName, data, isAppend) {
    const env = process.env["env"] ? process.env["env"] : "dev";
    const filePath = './data/' + env + '/' + datafileName
    try {
        const csvWriter = createObjectCsvWriter({
            path: filePath,
            header: Object.keys(data[0]).map((key) => ({id: key, title: key})),
            append: isAppend,
        });
        return csvWriter.writeRecords(data);
    } catch (error) {
        console.log('Error: ', error)
    }
}

File: backend/test-with-python/utils/load_csv.py
import csv
import os


def get_data_file_path(file_name):
    environment = os.environ.get('env') if os.environ.get('env') else "dev"
    current_folder = os.path.dirname(os.path.abspath(__file__))
    parent_folder = os.path.dirname(current_folder)
    file_path = parent_folder + '/data/' + environment + '/' + file_name
    return file_path


def read_csv(file_path):
    with open(file_path, 'r') as file:
        reader = csv.reader(file)
        data = list(reader)
    return data


def write_csv(file_path, data):
    with open(file_path, 'w', newline='') as file:
        writer = csv.writer(file)
        writer.writerows(data)
